tests/standalone/io/web_socket_error_test.dart

```diff
@@ -47,9 +47,10 @@ class SecurityConfiguration {
       ? HttpServer.bindSecure(HOST_NAME, 0, serverContext, backlog: backlog)
       : HttpServer.bind(HOST_NAME, 0, backlog: backlog);
 
-  Future<WebSocket> createClient(int port) =>
-      // TODO(whesse): Add a client context argument to WebSocket.connect.
-      WebSocket.connect('${secure ? "wss" : "ws"}://$HOST_NAME:$port/');
+  Future<WebSocket> createClient(int port) => secure
+      ? WebSocket.connect('${secure ? "wss" : "ws"}://$HOST_NAME:$port/',
+          customClient: HttpClient(context: clientContext))
+      : WebSocket.connect('${secure ? "wss" : "ws"}://$HOST_NAME:$port/');
```

Comment:
perhaps this can be simplified by passing `null` `customClient` if `secure` is `false`?
```
Future<WebSocket> createClient(int port) => 
      WebSocket.connect('${secure ? "wss" : "ws"}://$HOST_NAME:$port/',
          customClient: secure ? HttpClient(context: clientContext): null);

Reply:
completely agree, i wanted to keep the same logic as the server but didn't notice they were calling different functions.  I have the customUserAgent there for the added test.